### ISSUE TITLE
Parse error for nested declarations in selector-max-compound-selectors

### ISSUE THREAD

**Issue by maxmx** [stylelint/stylelint#2093]
In release 7.6.0 

The following sass which used to be valid in 7.5.0 now returns a `Cannot parse selector` error
```css
.foo {
  margin: {
    left: 5px;
    top: 10px;
  };
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
    "at-rule-name-case": "lower",
    "at-rule-name-space-after": "always-single-line",
    "at-rule-semicolon-newline-after": "always",
    "block-closing-brace-empty-line-before": "never",
    "block-closing-brace-newline-after": "always",
    "block-closing-brace-newline-before": "always-multi-line",
    "block-closing-brace-space-before": "always-single-line",

    "block-opening-brace-newline-after": "always-multi-line",
    "block-opening-brace-space-after": "always-single-line",
    "block-opening-brace-space-before": "always",
    "color-hex-case": "lower",
    "color-no-invalid-hex": true,
    "comment-empty-line-before": [ "always", {
      except: ["first-nested"],
      ignore: ["stylelint-commands"],
    } ],
    "comment-no-empty": true,
    "comment-whitespace-inside": "always",
    "custom-property-empty-line-before": [ "always", {
      except: [
        "after-custom-property",
        "first-nested",
      ],
      ignore: [
        "after-comment",
        "inside-single-line-block",
      ],
    } ],
    "declaration-bang-space-after": "never",
    "declaration-bang-space-before": "always",
    "declaration-block-no-duplicate-properties": true,
    "declaration-block-no-ignored-properties": true,
    "declaration-block-no-redundant-longhand-properties": true,
    "declaration-block-no-shorthand-property-overrides": true,
    "declaration-block-semicolon-newline-after": "always-multi-line",
    "declaration-block-semicolon-space-after": "always-single-line",
    "declaration-block-semicolon-space-before": "never",
    "declaration-block-single-line-max-declarations": 1,
    "declaration-block-trailing-semicolon": "always",
    "declaration-colon-newline-after": "always-multi-line",
    "declaration-colon-space-after": "always-single-line",
    "declaration-colon-space-before": "never",

    "font-family-name-quotes": "always-where-required",
    "function-calc-no-unspaced-operator": true,
    "function-comma-newline-after": "always-multi-line",
    "function-comma-space-after": "always-single-line",
    "function-comma-space-before": "never",
    "function-linear-gradient-no-nonstandard-direction": true,
    "function-max-empty-lines": 0,
    "function-name-case": "lower",
    "function-parentheses-newline-inside": "always-multi-line",
    "function-parentheses-space-inside": "never-single-line",
    "function-whitespace-after": "always",
    "indentation": 2,
    "keyframe-declaration-no-important": true,
    "media-feature-colon-space-after": "always",
    "media-feature-colon-space-before": "never",
    "media-feature-name-case": "lower",
    "media-feature-name-no-unknown": true,
    "media-feature-no-missing-punctuation": true,
    "media-feature-parentheses-space-inside": "never",
    "media-feature-range-operator-space-after": "always",
    "media-feature-range-operator-space-before": "always",
    "media-query-list-comma-newline-after": "always-multi-line",
    "media-query-list-comma-space-after": "always-single-line",
    "media-query-list-comma-space-before": "never",
    "no-empty-source": true,
    "no-eol-whitespace": true,
    "no-extra-semicolons": true,
    "no-invalid-double-slash-comments": true,
    "no-missing-end-of-source-newline": true,
    "property-case": "lower",
    "property-no-unknown": true,
    "rule-nested-empty-line-before": [ "always-multi-line", {
      except: ["first-nested"],
      ignore: ["after-comment"],
    } ],
    "rule-non-nested-empty-line-before": [ "always-multi-line", {
      ignore: ["after-comment"],
    } ],
    "selector-attribute-brackets-space-inside": "never",
    "selector-attribute-operator-space-after": "never",
    "selector-attribute-operator-space-before": "never",

    "selector-class-pattern": "^(?!js-).*",

    "selector-combinator-space-after": "always",
    "selector-combinator-space-before": "always",
    "selector-descendant-combinator-no-non-space": true,

    "selector-list-comma-space-before": "never",
    "selector-max-empty-lines": 0,
    "selector-pseudo-class-case": "lower",
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-class-parentheses-space-inside": "never",
    "selector-pseudo-element-case": "lower",
    "selector-pseudo-element-colon-notation": "single",
    "selector-pseudo-element-no-unknown": true,
    "selector-type-case": "lower",
    "selector-type-no-unknown": true,

    "string-no-newline": true,
    "unit-case": "lower",
    "unit-no-unknown": true,
    "value-list-comma-newline-after": "always-multi-line",
    "value-list-comma-space-after": "always-single-line",
    "value-list-comma-space-before": "never",
    "value-list-max-empty-lines": 0,
  }
```

> Which version of stylelint are you using?

e.g. `7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint 'src/**/*.scss'`

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
Cannot parse selector
```

**Comment by evilebottnawi** [comment]
@maxmx Thanks for feedback

**Comment by evilebottnawi** [comment]
@maxmx Today I see that is wrong :smile:

**Comment by jeddy3** [comment]
@maxmx Thanks for report and for using the template.

@evilebottnawi Thanks. Nothing in the changelog stands out to what might have caused this regression. Could it be related to [this refactoring](https://github.com/stylelint/stylelint/pull/1933)? Also, there [was a updated](https://github.com/stylelint/stylelint/commit/f7121f321ddc60c9b08f0c8519eafcc00113a2a1) of `postcss-scss` included in this release that might of triggered something.

**Comment by evilebottnawi** [comment]
@jeddy3 yep, i see this, maybe regression in `postcss-scss`, today will deal in this

**Comment by evilebottnawi** [comment]
Reproduce. Bug in `selector-max-compound-selectors`

Stack:
```shell
Error: Expected pseudo-class or pseudo-element
    at new error (/path/to/node_modules/postcss-selector-parser/dist/processor.js:29:23)
    at Parser.error (/path/to/node_modules/postcss-selector-parser/dist/parser.js:234:15)
    at Parser.pseudo (/path/to/node_modules/postcss-selector-parser/dist/parser.js:332:25)
    at Parser.parse (/path/to/node_modules/postcss-selector-parser/dist/parser.js:549:22)
    at Parser.loop (/path/to/node_modules/postcss-selector-parser/dist/parser.js:517:18)
    at new Parser (/path/to/node_modules/postcss-selector-parser/dist/parser.js:104:21)
    at Processor.process (/path/to/node_modules/postcss-selector-parser/dist/processor.js:26:21)
    at /path/to/node_modules/stylelint/dist/rules/selector-max-compound-selectors/index.js:69:14
    at Array.forEach (native)
    at /path/to/node_modules/stylelint/dist/rules/selector-max-compound-selectors/index.js:65:69
```

**Comment by davidtheclark** [comment]
Closed by #2102.
